lib/Twig/Extension/StringLoader.php

```diff
@@ -43,22 +43,5 @@ public function getName()
  */
 function twig_template_from_string(Twig_Environment $env, $template)
 {
-    $name = sprintf('__string_template__%s', hash('sha256', uniqid(mt_rand(), true), false));
-
-    $loader = new Twig_Loader_Chain(array(
-        new Twig_Loader_Array(array($name => $template)),
-        $current = $env->getLoader(),
-    ));
-
-    $env->setLoader($loader);
-    try {
-        $template = $env->loadTemplate($name);
-    } catch (Exception $e) {
-        $env->setLoader($current);
-
-        throw $e;
-    }
-    $env->setLoader($current);
-
-    return $template;
+    return $env->createTemplate($template);
```

Comment:
you could even save a method call in the compiled template by calling the env method directly instead of using a wrapper function